src/backend/distributed/utils/colocation_utils.c

```diff
@@ -96,6 +98,74 @@ mark_tables_colocated(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * update_distributed_table_colocation updates the colocation of a table.
+ * if colocate_with -> 'none' then the table is assigned a new
+ * colocation group.
+ */
+Datum
+update_distributed_table_colocation(PG_FUNCTION_ARGS)
+{
+	Oid destinationRelationId = PG_GETARG_OID(0);
+	text *colocateWithTableNameText = PG_GETARG_TEXT_P(1);
+
+	CheckCitusVersion(ERROR);
+	EnsureCoordinator();
+	EnsureTableOwner(destinationRelationId);
+	EnsureDistributedTable(destinationRelationId,
+						   "can update colocation of only a distributed table.");
```

Comment:
Passing the error message as a parameter seems like a strange pattern since we cannot easily add the table in question to the message.

Reply:
I updated the error message as `Relation should be a hash distributed table`.  Maybe it could be:

`Relation relation_name should be ... `